vpc/service/allocate.go

```diff
@@ -412,3 +416,178 @@ func (vpcService *vpcService) ValidateAllocation(ctx context.Context, req *titus
+			SubnetId:         aws.String(alloc.subnet),
+			Ipv6AddressCount: aws.Int64(1),
+		})
+		if err != nil {
+			return err
+		}
+
+		staticInterfaceID = createNetworkInterfaceOutput.NetworkInterface.NetworkInterfaceId
+		v6addrEni := createNetworkInterfaceOutput.NetworkInterface.Ipv6Addresses[0]
+		v6addr = net.ParseIP(aws.StringValue(v6addrEni.Ipv6Address))
+	}
+
+	v6output, err := session.CreateSubnetCidrReservation(ctx, ec2.CreateSubnetCidrReservationInput{
+		Cidr:            aws.String(fmt.Sprintf("%s/128", v6addr.String())),
+		Description:     aws.String(staticReservationDescription),
+		ReservationType: aws.String("explicit"),
+		SubnetId:        aws.String(alloc.subnet),
+	})
+	if err != nil {
+		return fmt.Errorf("Cannot make reservation for IPv6 addr: %w", err)
+	}
+
+	v4output, err := session.CreateSubnetCidrReservation(ctx, ec2.CreateSubnetCidrReservationInput{
+		Cidr:            aws.String(fmt.Sprintf("%s/32", v4addr.String())),
+		Description:     aws.String(staticReservationDescription),
+		ReservationType: aws.String("explicit"),
+		SubnetId:        aws.String(alloc.subnet),
+	})
+	if err != nil {
+		return fmt.Errorf("Cannot make reservation for IPv4 addr: %w", err)
+	}
+
+	_, err = tx.ExecContext(ctx,
+		"INSERT INTO subnet_cidr_reservations_v6(reservation_id, subnet_id, prefix, type, description) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING ",
+		aws.StringValue(v6output.SubnetCidrReservation.SubnetCidrReservationId),
+		alloc.subnet,
+		aws.StringValue(v6output.SubnetCidrReservation.Cidr),
+		aws.StringValue(v6output.SubnetCidrReservation.ReservationType),
+		aws.StringValue(v6output.SubnetCidrReservation.Description),
+	)
+	if err != nil {
+		return fmt.Errorf("Could not insert subnet CIDR v6 reservation %s: %w", aws.StringValue(v6output.SubnetCidrReservation.SubnetCidrReservationId), err)
+	}
+
+	_, err = tx.ExecContext(ctx,
+		"INSERT INTO subnet_cidr_reservations_v4(reservation_id, subnet_id, prefix, type, description) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING ",
+		aws.StringValue(v4output.SubnetCidrReservation.SubnetCidrReservationId),
+		alloc.subnet,
+		aws.StringValue(v4output.SubnetCidrReservation.Cidr),
+		aws.StringValue(v4output.SubnetCidrReservation.ReservationType),
+		aws.StringValue(v4output.SubnetCidrReservation.Description),
+	)
+	if err != nil {
+		return fmt.Errorf("Could not insert subnet CIDR v4 reservation %s: %w", aws.StringValue(v4output.SubnetCidrReservation.SubnetCidrReservationId), err)
+	}
+
+	_, err = tx.ExecContext(ctx,
+		"UPDATE ip_addresses SET v4prefix = $1, v6prefix = $2 WHERE id = $3",
+		aws.StringValue(v4output.SubnetCidrReservation.SubnetCidrReservationId),
+		aws.StringValue(v6output.SubnetCidrReservation.SubnetCidrReservationId),
+		alloc.id,
+	)
+	if err != nil {
+		return fmt.Errorf("Could not update prefixes: %w", err)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("Could not commit transaction: %w", err)
+	}
+
+	if staticInterfaceID != nil {
```

Comment:
A TODO is to add back the code to delete these.

Reply:
added